opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -157,12 +167,25 @@ class SpanContext:
     def __init__(self,
                  trace_id: int,
                  span_id: int,
-                 options: 'TraceOptions',
-                 state: 'TraceState') -> None:
+                 traceoptions: 'TraceOptions' = None,
+                 tracestate: 'TraceState' = None
+                 ) -> None:
+        if traceoptions is None:
```

Comment:
According to the [current specification](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#spancontext), we need to have `traceOptions`.

Reply:
Or `trace_options`, either way looks weird though.

Reply:
I don't think the spec is meant to dictate internal names like this, especially since it's clear what this field represents. In any case we'll follow the W3C spec when the class gets serialized.

Reply:
I think it's more about "do we treat traceoptions as a single word or not" rather than "do we take the name strictly from the spec".
The W3C spec is using two words, having `traceoptions` is against Python naming convention.

Reply:
That's fair. I prefer `traceoptions/state` but agree it's more consistent to split the words up. Changed in 0f52732.